### PR TITLE
support server

### PR DESCRIPTION
**Pull request by psladek**
support server and supportserver generator implementation

### CODE REVIEW

File: tests/support_server/setup.pm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+       setup_pxe_server();
+       setup_tftp_server();
+    }
+    if ( exists $server_roles{'qemuproxy'} ) {    
+       #setup pxe server
+       setup_http_server();
+       $setup_script.="curl -f -v " . autoinst_url . "/data/supportserver/proxy.conf | sed -e 's|#AUTOINST_URL#|" . autoinst_url . "|g' >/etc/apache2/vhosts.d/proxy.conf\n";
+       $setup_script.="systemctl restart apache2.service\n";
+
+    }
```
Comment (aaannz) [inline tests/support_server/setup.pm]: This many empty lines are really distracting for me, sorry.
Apart of that I see what are you trying to do, but I'm confused
- by empty subroutines like setup_nfs_mount, setup_ftp_server
- the *#setup pxe server* comment on L107, why is it there? Isn't something missing? Also I think this comment on L102 is redundant given the function call.

File: tests/support_server/wait.pm [hunk 2]
```diff
@@ -21,25 +20,9 @@ use lockapi;
...
-    type_password;
-    type_string "\n";
-    sleep 5;
-    
-    
-    script_output("
-        curl -f -v " . autoinst_url . "/data/supportserver/proxy.conf | sed -e 's|#AUTOINST_URL#|" . autoinst_url . "|g' >/etc/apache2/vhosts.d/proxy.conf
-        systemctl restart apache2.service
-    ");
-
```
Comment (aaannz) [inline tests/support_server/wait.pm]: I would add at least *wait_idle* for the peace of the mind, and/or comment that this really should be loaded after **support_server/setup.pm**. 
The name of the test **wait.pm** is not very descriptive for other IMO, what about **wait_for_children.pm**?

File: tests/support_server/wait.pm [hunk 3]
```diff
@@ -66,5 +53,3 @@ sub test_flags {
 }
 
 1;
-
-# vim: set sw=4 et:
```
Comment (aaannz) [inline tests/support_server/wait.pm]: Not a fan of vim, right? :)

File: tests/support_server/wait.pm (hunk 2 above)
Comment (psladek) [inline tests/support_server/wait.pm]: does wait idle really help if there is no continuous change anyway?
wait_for_children.pm is a bit long imho, especially in comparison with other supportserver module names

Comment (psladek) [inline tests/support_server/wait.pm]: there is also significat time before the dependent machine start, but ok, maybe wait idle is safer. Will check if it does not delay support server setup too much

File: tests/autoyast/pxe_boot.pm [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Copyright (C) 2014 SUSE Linux Products GmbH
+# Copyright (C) 2015 SUSE Linux GmbH
```
Comment (sysrich) [inline tests/autoyast/pxe_boot.pm]: Does this test module ever get used at all? I cant find anywhere that calls it..

Comment (psladek) [inline tests/autoyast/pxe_boot.pm]: yes, it is used if PXEBOOT is set, at least on sle-11-SP3

Comment (sysrich) [inline tests/autoyast/pxe_boot.pm]: can you show me where? I'm doing my best to search but I can't find it..

Comment (psladek) [inline tests/autoyast/pxe_boot.pm]: hm, strange, it should be in tests/sle-11-SP3/main.pm, on line cca 335

**Pull request merged**